### PR TITLE
UI/UX improvements to coordinate cursor lock button and stylus/freehand

### PR DESCRIPTION
**Pull request by nirvn**
The coordinate cursor lock button has always been IMHO a bit unfriendly. This PR aims at improving things in that regard. GIF:
Peek 2020-12-24 17-53

Visual improvements include:
- A clearer disabled state by adopting a semi-opaque gray background (which we now use consistently to indicate a deactivated toggle button). _Bonus: more of the canvas is visible when the button is toggled off :tada: _
- A clearer activated style by using a blue highlight for the  icon itself (in addition to the background going fully opaque)
- A clearer inactive and active icon; for inactive, it simply looks like the coordinate cross while for active, we get a clearer sense of the coordinate cross being locked in / boxed in the location 
- The addition of a toaster message to indicate to the user what this button is all about :wink:

Experience improvements include:
- If the user toggles the lock on and the position is outside the current map bounding box, we now automatically re-center. There's no point in having this on while not being able to see where you are upon activation
- The X/Y/length/area information text now follows the location of the cursor
- When the cursor is locked, the stylus hover is disabled which means the (+) button stays visible and the stylus can be used to hit that button

The  information text following the cursor is a nice improvement to the stylus / freehand too:
Peek 2020-12-24 17-31

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for ce983e7e7a498fdb44d6d7e596cd71b24bbd0029:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-feb3cea1f9e3a5529c55542e4f821ba9b980add1-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-feb3cea1f9e3a5529c55542e4f821ba9b980add1-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-feb3cea1f9e3a5529c55542e4f821ba9b980add1-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-feb3cea1f9e3a5529c55542e4f821ba9b980add1-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 87d004e19657ee3133ec550c020227a26be597d0:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-191cf72938e3468417b1723976b5ccaa8f804165-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-191cf72938e3468417b1723976b5ccaa8f804165-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-191cf72938e3468417b1723976b5ccaa8f804165-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-191cf72938e3468417b1723976b5ccaa8f804165-x86.apk)

File: src/qml/qgismobileapp.qml [hunk 1]
```diff
@@ -555,8 +555,18 @@ ApplicationWindow {
       crs: qgisProject.crs
     }
 
-    x: mapCanvas.width / 2 + 10
-    y: mapCanvas.height / 2 + 22
+    x: {
```
Comment (m-kuhn) [inline src/qml/qgismobileapp.qml:558]: I think that deservs a short explsnation comment

**Review by m-kuhn — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 60c7c63af852bf8fb091357981343e1fc39f7881:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-990a36f000ea766dda8aabac7f0014d7068d5bb8-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-990a36f000ea766dda8aabac7f0014d7068d5bb8-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-990a36f000ea766dda8aabac7f0014d7068d5bb8-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1520-990a36f000ea766dda8aabac7f0014d7068d5bb8-x86.apk)

**Pull request merged**